### PR TITLE
`General`: Elevate toggle sidebar button

### PR DESCRIPTION
**Pull request by PaRangger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Currently when the sidebar is collapsed the users, especially new ones, might not easily find the button to show it again. This change should elevate the button more to make it more noticable.

### Description

I changed the color of the button, as well as added a chevron to indicate the sidebar can be unhidden. Additionally to be more prominent I added a small animation.

### Steps for Testing

Prerequisites:
- 1 Student/Tutor/Instructor
- 1 Course with a Lecture/Course or Communication enabled

1. Log in to Artemis
2. Navigate to the course
3. Locate the sidebar button and click it
4. If the sidebar is hidden the chevron point right. If the sidebar is shown the chevron should point left. In both cases the color should change.
5. Make sure that when navigating between exercises/lectures/communication that the button reacts properly to the sidebar being shown or not shown.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

### Screenshots

Bildschirmfoto 2024-10-02 um 14 22 15
Bildschirmfoto 2024-10-02 um 14 22 24

## Summary by CodeRabbit

- **New Features**
	- Enhanced sidebar functionality with a toggle button that visually indicates its state.
	- Conditional display of the sidebar toggle button based on the course view.

- **Improvements**
	- Improved transition effects for the sidebar collapse button, enhancing user experience.
	- Updated tooltip for clearer instructions on toggling the sidebar.

- **Visual Enhancements**
	- Added new styles for smoother animations and visual feedback during sidebar interactions.

### CODE REVIEW

File: src/main/webapp/app/overview/course-overview.component.html [hunk 1]
```diff
@@ -146,11 +146,16 @@
                             @if (hasSidebar) {
                                 <button
                                     [ngbTooltip]="'Toggle Sidebar (⇧ + Ctrl + B)'"
+                                    [ngClass]="isSidebarCollapsed ? 'is-collapsed' : ''"
```
Comment (rabeatwork) [inline src/main/webapp/app/overview/course-overview.component.html:149]:
```suggestion
                                    [ngClass]="{'is-collapsed' : isSidebarCollapsed}"
```

**Review by rabeatwork — CHANGES_REQUESTED** [review]
I have one small improvement for the code and a general remark. The background color transition also affects the hover effect transition, causing the hover background color to appear with a delay.

File: src/main/webapp/app/overview/course-overview.component.html (hunk 1 above)
Comment (PaRangger) [reply src/main/webapp/app/overview/course-overview.component.html]: Thanks! In the last commits I fixed the hover animation and added the suggested change 👍🏼

**Comment by janthoXO** [comment]
for better usability the sidebar collapse buttons should be consistent. if you're are already touching the collapse buttons, please make them consistent.
The best approach would probably be to make the sidebar button for the exercises similar to the general sidebar as a >> button in the sidebar itself.
And maybe even move the buttons to the top of each sidebar

**Comment by krusche** [comment]
> for better usability the sidebar collapse buttons should be consistent. if you're are already touching the collapse buttons, please make them consistent. The best approach would probably be to make the sidebar button for the exercises similar to the general sidebar as a >> button in the sidebar itself. And maybe even move the buttons to the top of each sidebar

@rabeatwork put a lot of thoughts in the design of the sidebar. It is explicitly designed to hide completely to maximize the space used to show actual content.  We can certainly discuss this in a separate meeting how to further improve the user experience of the sidebar. 

However, this intentionally small PR focuses on an immediate solution to a very specific usability problem: a user might not realize how to find an exercise / lecture / channel / etc. when they open a view with a collapsed sidebar. Let's fix this issue first and then have a discussion about additional changes.

**Review by krusche — APPROVED** [review]

**Pull request merged**